tests/functional/tests/engine/test_flush.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright(c) 2022-2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+from ctypes import c_int
+
+from pyocf.types.cache import Cache
+from pyocf.types.data import Data
+from pyocf.types.core import Core
+from pyocf.types.io import IoDir
+from pyocf.types.volume import Volume, IoFlags, TraceDevice
+from pyocf.utils import Size
+from pyocf.types.shared import OcfCompletion
+
+
+def test_flush_propagation(pyocf_ctx):
+    flushes = {}
+
+    pyocf_ctx.register_volume_type(TraceDevice)
+
+    def trace_flush(vol, io, io_type):
+        nonlocal flushes
+
+        if io_type == TraceDevice.IoType.Flush:
+            if not vol.uuid in flushes:
+                flushes[vol.uuid] = []
+            flushes[vol.uuid].append((io.contents._addr, io.contents._bytes))
+
+        return True
```

Comment:
Is this intended to always return True? `TraceDevice._trace()` seem to rely on the return value from this `trace_flush()`.

Reply:
Return from trace function decides whether or not to let the IO through, in this case I think Adam didn't mean to inject any errors so `return True` seems to be apropriate